WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java

```diff
@@ -216,12 +251,106 @@ public void onResponse(JSONObject response) {
     }
 
+    private void pushWpSettings() {
+        try {
+            final JSONObject jsonParams = serializeDotComParamsToJSONObject();
+            // skip network requests if there are no changes
+            if (jsonParams.length() <= 0) {
+                return;
+            }
+            WordPress.getRestClientUtilsV1_1().setGeneralSiteSettings(
+                    mSite.getSiteId(), jsonParams, new RestRequest.Listener() {
+                        @Override
+                        public void onResponse(JSONObject response) {
+                            AppLog.d(AppLog.T.API, "Site Settings saved remotely");
+                            notifySavedOnUiThread(null);
+                            mRemoteSettings.copyFrom(mSettings);
+
+                            if (response != null) {
+                                JSONObject updated = response.optJSONObject("updated");
+                                if (updated == null) return;
+                                HashMap<String, Object> properties = new HashMap<>();
+                                Iterator<String> keys = updated.keys();
+                                while (keys.hasNext()) {
+                                    String currentKey = keys.next();
+                                    Object currentValue = updated.opt(currentKey);
+                                    if (currentValue != null) {
+                                        properties.put(SAVED_ITEM_PREFIX + currentKey, currentValue);
+                                    }
+                                }
+                                AnalyticsUtils.trackWithSiteDetails(
+                                        AnalyticsTracker.Stat.SITE_SETTINGS_SAVED_REMOTELY, mSite, properties);
+                            }
+                        }
+                    }, new RestRequest.ErrorListener() {
+                        @Override
+                        public void onErrorResponse(VolleyError error) {
+                            AppLog.w(AppLog.T.API, "Error POSTing site settings changes: " + error);
+                            notifySavedOnUiThread(error);
+                        }
+                    });
+        } catch (JSONException exception) {
+            AppLog.w(AppLog.T.API, "Error serializing settings changes: " + exception);
+            notifySavedOnUiThread(exception);
+        }
+    }
+
+    private void pushJetpackProtectAndSsoSettings() {
+        final Map<String, Object> params = serializeJetpackProtectAndSsoParams();
+        if (params.isEmpty()) {
+            return;
+        }
+        WordPress.getRestClientUtilsV1_1().setJetpackSettings(mSite.getSiteId(), params,
+                new RestRequest.Listener() {
+                    @Override
+                    public void onResponse(JSONObject response) {
+                        AppLog.d(AppLog.T.API, "Jetpack settings updated");
+                        mRemoteJpSettings.monitorActive = mJpSettings.monitorActive;
```

Comment:
In that case, I think we should just copy the values in `mJpSettings` just before the call and use those values in the callback as there is no guarantee that `mJpSettings` is not modified during the call.